OpenEdXMobile/src/main/java/org/edx/mobile/util/DataBindingHelperUtils.kt

```diff
@@ -114,47 +123,109 @@ class DataBindingHelperUtils {
+                }
+                CourseDateType.PAST_DUE -> {
+                    badgeBackground = R.color.bullet_light_gray
+                    textAppearance = R.style.past_due_badge_text_appearance
+                }
+                CourseDateType.DUE_NEXT -> {
+                    badgeBackground = R.color.bullet_dark_gray
+                    textAppearance = R.style.due_next_badge_text_appearance
+                }
+                CourseDateType.NOT_YET_RELEASED -> {
+                    badgeBackground = android.R.color.transparent
+                    textAppearance = R.style.not_yet_released_badge_text_appearance
+                    badgeStrokeColor = R.color.tag_light_silver_border
+                }
+                else -> {
+                    return
+                }
+            }
+            addDateBadge(textView, courseDateType.getTitle(), badgeBackground, textAppearance, badgeIcon, badgeStrokeColor)
+            if (isToday && type == CourseDateType.COMPLETED) {
+                setBadge(textView, type, false, dateBlockItems)
+            }
+        }
+
+        private fun hasSameDateTypes(dateBlockItems: ArrayList<CourseDateBlock>?): Boolean {
+            if (dateBlockItems != null && dateBlockItems.isEmpty().not() && dateBlockItems.size > 1) {
+                val dateType = dateBlockItems[0].dateBlockBadge
+                for (i in 1 until dateBlockItems.size) {
+                    if (dateBlockItems[i].dateBlockBadge != dateType &&
+                            dateBlockItems[i].dateBlockBadge != CourseDateType.BLANK) {
+                        return false
                     }
                 }
             }
-            textView.visibility = View.VISIBLE
+            return true
+        }
+
+        /**
+         * Method to add the date badge at the end in given title with styles attributes
+         */
+        private fun addDateBadge(textView: TextView, title: String, badgeBackground: Int,
+                                 textAppearance: Int,
+                                 badgeIcon: IconDrawable?,
+                                 badgeStrokeColor: Int) {
+            // add badge title so can identify the actual badge position
+            val titleWithBadge = SpannableStringBuilder((textView.text as Spannable))
+            // add space before badge title
+            titleWithBadge.append("  $title")
+            textView.setText(titleWithBadge, TextView.BufferType.SPANNABLE)
+
+            // setup the date badge
+            val string = textView.text as Spannable
+            val chipDrawable = ChipDrawable.createFromResource(textView.context,
+                    R.xml.dates_badge_chip)
+            chipDrawable.text = title
+            chipDrawable.setChipBackgroundColorResource(badgeBackground)
+            chipDrawable.setTextAppearanceResource(textAppearance)
+            if (badgeIcon != null) {
+                chipDrawable.chipIcon = badgeIcon
```

Comment:
Please add chipIcon size and padding i.e,
`chipDrawable.chipIconSize = textView.context.resources.getDimension(R.dimen.small_icon_size)
                chipDrawable.setIconStartPaddingResource(R.dimen.dates_badge_icon_start_padding)
                chipDrawable.setIconEndPaddingResource(R.dimen.dates_badge_icon_end_padding)`

Reply:
added.